src/app/content/components/NudgeStudyTools/utils.ts

```diff
@@ -0,0 +1,168 @@
+}
+
+export const getPositions = (target: HTMLElement, isMobile: boolean, windowWidth: number): Positions => {
+  const { top, left, right, height, width } = getBoundingRectOfNudgeTarget(target);
+  const padding = remsToPx(spotlightPadding);
+  const spotlightTopOffset = top - padding;
+  const spotlightLeftOffset = left - padding;
+  const spotlightHeight = height + (padding * 2);
+  const spotlightWidth = width + (padding * 2);
+  const arrowTopOffset = spotlightTopOffset + spotlightHeight + remsToPx(arrowTopMargin);
+  const arrowLeft = spotlightLeftOffset + remsToPx(arrowLeftMargin);
+  const contentWrapperTopOffset = arrowTopOffset
+    + remsToPx(isMobile ? arrowMobileHeight : arrowDesktopHeight)
+    + remsToPx(contentMarginTop);
+  const contentWrapperRight = windowWidth - right - padding;
+  const closeButtonLeft = left + spotlightWidth;
+  const closeButtonTopOffset = contentWrapperTopOffset - remsToPx(closeButtonDistanceFromContent);
+
+  return {
+    arrowLeft,
+    arrowTopOffset,
+    closeButtonLeft,
+    closeButtonTopOffset,
+    contentWrapperRight,
+    contentWrapperTopOffset,
+    spotlightHeight,
+    spotlightLeftOffset,
+    spotlightTopOffset,
+    spotlightWidth,
+  };
+};
+
+// Target may have display set to `contents` so we are calculatings rect for children of passed target.
+const getBoundingRectOfNudgeTarget = (target: HTMLElement) => {
+  // Starting values depends on if we checks for Math.max or Math.min
+  let top = 9999;
+  let bottom = 0;
+  let left = 9999;
+  let right = 0;
+
+  for (const child of Array.from(target.children)) {
+    const rect = child.getBoundingClientRect();
+    top = Math.min(top, rect.top);
+    bottom = Math.max(bottom, rect.bottom);
+    left = Math.min(left, rect.left);
+    right = Math.max(right, rect.right);
+  }
+
+  return {
+    height: bottom - top,
+    left,
+    right,
+    top,
+    width: right - left,
+  };
+};
+
+export const usePositions = (isMobile: boolean) => {
+  const [windowWidth] = useDebouncedWindowSize();
+  const [positions, setPositions] = React.useState<Positions | null>(null);
+  const studyGuides = useSelector(hasStudyGuides);
+  const isEnabled = useSelector(studyGuidesEnabled);
+
+  React.useEffect(() => {
+    const document = assertDocument();
+    const target = document.querySelector(`#${nudgeStudyToolsTargetId}`) as HTMLElement | null;
+    if (isEnabled && target) {
+      // Make sure that we calculate positions with body overflow set to hidden
+      // because it causes scrollbar to hide which results with different positions.
+      const prevOverflow = document.body.style.overflow;
+      document.body.style.overflow = 'hidden';
+      setPositions(getPositions(target, isMobile, windowWidth));
+
+      // Resets to the value from before calculations. We want this style change to be handled
+      // directly in the component.
+      document.body.style.overflow = prevOverflow;
+    }
+    return () => setPositions(null);
```

Comment:
What problem does this solve?

Reply:
It makes sure that incorrect data is not stored between rerenders and it is cleared if target is no longer in the dom

Reply:
You're correct. Since this is sync there should be no case when this data is outdated
However, using this seems cleaner for me than `else` statement
I think there should be no real difference between those two